seatunnel-engine/seatunnel-engine-storage/imap-storage-plugins/imap-storage-file/src/main/java/org/apache/seatunnel/engine/imap/storage/file/disruptor/WALWorkHandler.java

```diff
@@ -78,6 +78,9 @@ private void walEvent(IMapFileData iMapFileData, WALEventType type, long request
     }
 
     private void executeResponse(long requestId, boolean success) {
+        if (null == RequestFutureCache.get(requestId)) {
+            log.warn("requestId is {} not found in RequestFutureCache", requestId);
+        }
```

Comment:
It can also cause a NPE if null . If it's null, maybe don't have to call `RequestFutureCache.get(requestId).done(success);`

Reply:
good cache~